### PR TITLE
NAS-122357 / 23.10 / Fix adding static entries to /etc/hosts

### PR DESCRIPTION
**Pull request by yocalebo**
Many problems found while investigating this issue. For one, we added a file named `global.py`.... `global` is a keyword in python so when I added the `HOSTS_FILE_EARMARKER` global variable, I was unable to import it into the mako file because of a syntax error. For two, adding static host table entries to /etc/hosts and having them take effect immediately has been broken on SCALE for a year or more minimally. It has required a restart before those entries have been added to /etc/hosts. Lastly, we made the hosts key in the schema a string which is incredibly hard to validate in any meaningful way. I've changed this to a list object (just like we do for the search domains). This allows us to more easily (and in a more standard programmatic way) validate the entries given to us by the end-user.

To summarize the improvements/fixes:
- remove unnecessary newlines in global_config.py (too many of them)
- simplify the logic (and also fix a bug) when validating the `nameserver*` keys
- fix the initial bug where we were not regenerating /etc/hosts on a running system (without these changes, it required a reboot)
- rename `network_/global.py` to `network_/global_config.py`
- update api tests to test new functionality

### CODE REVIEW

File: tests/api2/test_003_network_global.py [hunk 1]
```diff
@@ -59,10 +61,28 @@ def test_001_set_and_verify_network_global_settings_database(ws_client, netinfo)
...
-def test_002_verify_network_general_summary(request, ws_client, netinfo, ip_to_use):
+def test_003_verify_network_general_summary(request, ws_client, netinfo, ip_to_use):
     depends(request, ['NET_CONFIG'])
     summary = ws_client.call('network.general.summary')
-    assert set(summary['nameservers']) == set([netinfo[i] for i in netinfo if i.startswith('nameserver')])
-    assert summary['default_routes'][0] == netinfo['ipv4gateway']
     assert any(i.startswith(ip_to_use) for i in summary['ips'][interface]['IPV4'])
+
+
+def test_004_exit_early():
```
Comment (themylogin) [inline tests/api2/test_003_network_global.py:87]: Leftover
